MennaMohie/assignment2/Solution.cpp

```diff
@@ -12,118 +14,111 @@ class Node
     friend class BinaryTree<T>;
 
 public:
-    Node():left(nullptr),right(nullptr) {}
     Node(T val):left(nullptr),right(nullptr),value(val) {}
 
-    T getValue(){return value;}
+    T getValue()
+    {
+        return value;
+    }
 };
 
 template<class T>
 class BinaryTree
 {
     Node<T> *root;
-
-public:
-    BinaryTree(T val):root(new Node<T>(val)) {}
-
-    Node<T>* getRoot()
-    {
-        return root;
-    }
-
-    bool contains(Node<T> *currentRoot, T val)
-    {
-        if(currentRoot == nullptr)
-            return false;
-        if(currentRoot->value == val)
-            return true;
-        return (contains(currentRoot->left, val) || contains(currentRoot->right, val));
-    }
-
-    bool resetRoot(T val)
+    int Size;
+    add(Node<T>* currentRoot, T parentVal, T val, direction dir)
     {
-        if(root==nullptr)
-        {
-            root=new Node<T>(val);
-            return true;
-        }
-        else if(!contains(root, val))
-        {
-            root->value=val;
-            return true;
-        }
-        else
-            return false;
-    }
-
-    //adds a new node (val) as a left child to an existing node (parentVal)
-    bool addAsLeftChild(Node<T> *currentRoot, T parentVal, T val)
-    {
-        if(!contains(root, parentVal) || contains(root, val) || currentRoot==nullptr)
-            return false;
         if(currentRoot->value==parentVal)
         {
-            if(currentRoot->left==nullptr)
+            if(dir==0)
```

Comment:
Since you have the enum, better to use it :)
`dir == addLeft` looks better, right?